src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunner.java

```diff
@@ -0,0 +1,93 @@
+package org.jetbrains.bsp.bazel.server.bazel;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Semaphore;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.bsp.bazel.server.bazel.data.ProcessResults;
+
+public class BazelRunner {
+
+  private static final Logger LOGGER = LogManager.getLogger(BazelRunner.class);
+
+  private static final String PUBLISH_ALL_ACTIONS = "--build_event_publish_all_actions";
+  private static final String BES_BACKEND = "--bes_backend=grpc://localhost:";
+
+  private final Semaphore processLock;
+  private final String bazel;
+  private Integer besBackendPort;
```

Comment:
Perhaps we could use an [Optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) to make sure the access to this variable is safe. Wdyt?